### ISSUE TITLE
Breaks DateTime string

### ISSUE THREAD

**Issue by lcherone** [camspiers/json-pretty#9]
If there is a datetime value in the json, it will break the value and cause `2015-09-08 08:56:10` to become `2015-09-08 08: 56: 10`

 - Line 71,
   https://github.com/camspiers/json-pretty/blob/master/src/Camspiers/JsonPretty/JsonPretty.php#L71

There are other issues with commas

Using **JSON_PRETTY_PRINT**

    [
        {
            "id": "34",
            "system_id": "2000",
            "src": "***",
            "dst": "790",
            "dcontext": "RingGroups-2000",
            "clid": "***",
            "channel": "SIP\/***-0000005c",
            "dstchannel": "SIP\/2000200-0000005d",
            "lastapp": "Dial",
            "lastdata": "SIP\/2000200&SIP\/2000201),20,ktwx",
            "calldate": "2015-09-08 08:56:10",
            "answer": "2015-09-08 08:56:10",
            "end": "2015-09-08 08:56:20",
            "duration": "10",
            "billsec": "0",
            "disposition": "NO ANSWER",
            "uniqueid": "pbx1-1441698970.92",
            "linkedid": "pbx1-1441698970.92",
            "internal": "0",
            "outbound": "0",
            "inbound": "1",
            "uploaded": "0",
            "ddi": null
        }
    ]

Using your lib:

    [
            {
                    "id": "34",
                    "system_id": "2000",
                    "src": "***",
                    "dst": "790",
                    "dcontext": "RingGroups-2000",
                    "clid": "***",
                    "channel": "SIP\/***-0000005c",
                    "dstchannel": "SIP\/2000200-0000005d",
                    "lastapp": "Dial",
                    "lastdata": "SIP\/2000200&SIP\/2000201),
                    20,
                    ktwx",
                    "calldate": "2015-09-08 08: 56: 10",
                    "answer": "2015-09-08 08: 56: 10",
                    "end": "2015-09-08 08: 56: 20",
                    "duration": "10",
                    "billsec": "0",
                    "disposition": "NO ANSWER",
                    "uniqueid": "pbx1-1441698970.92",
                    "linkedid": "pbx1-1441698970.92",
                    "internal": "0",
                    "outbound": "0",
                    "inbound": "1",
                    "uploaded": "0",
                    "ddi": null
            }
    ]

**Comment by camspiers** [comment]
Looks like the inString tracking is broken, because it thinks that it isn't in a string when it actually is.

**Comment by camspiers** [comment]
Hey @h4cc I think that your latest change may have broken the inString tracking:

You added:

```php
            } elseif ($char === '"') {
                if (
                    // A String is ending, when there is a not escaped quote ...
                    ($c > 0 && $json[$c - 1] !== '\\')
                    &&
                    // and a String is ending, when there is a quote prepended with a escaped slash.
                    ($c > 1 && $json[$c - 2].$json[$c - 1] === '\\\\')
                ) {
                    $inString = !$inString;
                }
                $result .= $char;
```

Do you think that it is meant to be:

```php
            } elseif ($char === '"') {
                if (
                    // A String is ending, when there is a not escaped quote ...
                    ($c > 0 && $json[$c - 1] !== '\\')
                    ||
                    // and a String is ending, when there is a quote prepended with a escaped slash.
                    ($c > 1 && $json[$c - 2].$json[$c - 1] === '\\\\')
                ) {
                    $inString = !$inString;
                }
                $result .= $char;
```

**Comment by h4cc** [comment]
Will add a test that covers this.

**Comment by camspiers** [comment]
Resolved by 67a5377, release available using the 1.0.1 tag
